source/client.go

```diff
@@ -83,3 +86,30 @@ func (c Client) Update(id string, params *stripe.SourceObjectParams) (*stripe.So
 func getC() Client {
 	return Client{stripe.GetBackend(stripe.APIBackend), stripe.Key}
 }
+
+// Detach detaches the source from its customer object.
+func Detach(id string, params *stripe.SourceObjectParams) (*stripe.Source, error) {
+	return getC().Detach(id, params)
+}
+
+func (c Client) Detach(id string, params *stripe.SourceObjectParams) (*stripe.Source, error) {
+	var body *form.Values
+	var commonParams *stripe.Params
+
+	if params != nil {
+		commonParams = &params.Params
+		body = &form.Values{}
+		form.AppendTo(body, params)
+	}
+
+	source := &stripe.Source{}
+	var err error
+
+	if len(params.Customer) > 0 {
+		err = c.B.Call("DELETE", fmt.Sprintf("/customers/%v/sources/%v", params.Customer, id), c.Key, body, commonParams, source)
```

Comment:
Since `SourceObjectParams` has it defined as `Customer string             `form:"customer"`` won't the parameter be posted anyway to the API? I guess you might need to create a `SourceDetachParams` instead to avoid this issue.

Reply:
Good point, though that probably also applies to the `Del` method on cards and bank accounts then. I'll check!

Reply:
Okay, you were entirely correct. Cards and bank accounts aren't affected since `Customer` has `form:"-"` in their respective structures.

Fixed by defining a new `SourceObjectDetachParams` struct.